src/AnimalFamily/AnimalFamily.ts

```diff
@@ -22,56 +22,57 @@ export default class AnimalFamily {
       requirements,
       animals[0]
     );
-    const verifycationResult: VerificationResult =
+    const verificationResult: VerificationResult =
       verifyFamilyInfo(specificRequirements);
-    if (verifycationResult.message === "") {
+    if (verificationResult.message === "") {
       this.name = name;
       animals.forEach((animal) => {
         animal.setHome(this.name);
       });
       this.animals = animals;
       families.push(this);
     } else {
-      throw new Error(verifycationResult.message);
+      throw new Error(verificationResult.message);
     }
   }
-  addAnimal(animal: Animal, fromBirth = false) {
+  //return an empty string if it can be added otherwise the reason that it can't be added
+  checkCanAddAnimal(animal: Animal): string {
```

Comment:
add access modifiers to the new methods, also I think maybe `checkCanAddAnimal` and `checkCanRemoveAnimal` should be private since they are only supposed to be used inside this class

Reply:
I don't think they can be private since i am using them in the switchFamily method in the animal class